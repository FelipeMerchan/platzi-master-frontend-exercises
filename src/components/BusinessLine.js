import React from 'react'

import './styles/BusinessLine.css'

function BusinessLine(props) {
  return (
    <article className="BusinessLine">
      <figure className="BusinessLine__imageContainer">
        <img className="BusinessLine__image" width="200" src={props.image} alt={props.alt} />
      </figure>
      <div>
        <h3 className="BusinessLine__name text-consultancy-color" style={props.style} >{props.title}</h3>
        <p className="BusinessLine__description">
          Lorem ipsum dolor sit amet, consectetuer aiscing elit.
        </p>
      </div>
    </article>
  )
}

export default BusinessLine