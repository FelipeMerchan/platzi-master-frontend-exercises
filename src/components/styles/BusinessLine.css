.BusinessLine {
  border: 2px solid #dfdfdf;
  background-color: var(--surfaceColor);
  font-family: var(--font-3);
  color: #a6a7a2;
}

.BusinessLine__image {
  width: 100%;
  height: 300px;
  object-fit: cover;
}

.BusinessLine > div {
  min-height: 170px;
  padding: var(--space-2);
}

.BusinessLine__name {
  margin: 0 0 var(--space);
  padding-bottom: var(--space);
  padding-left: var(--space);
  border-bottom: 1px solid #dfdfdf;
  font-size: 28px;
}

.BusinessLine__description {
  margin-bottom: 0;
  font-size: 16px;
  color: #a6a7a2;
}