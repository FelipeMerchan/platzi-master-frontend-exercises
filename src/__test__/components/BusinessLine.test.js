import React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import '../../setupTest'
import BusinessLine from '../../components/BusinessLine'

describe('Probar el componente <BusinessLine />', () => {
  const businessLine = shallow(<BusinessLine />)
  test('Render del componente BusinessLine', () => {
    expect(businessLine.length).toEqual(1)
  })
  test('Render de la descripción de la línea de negocio', () => {
    expect(businessLine.find('.BusinessLine__description').text()).toEqual('Lorem ipsum dolor sit amet, consectetuer aiscing elit.')
  })
})

describe('BusinessLine Snapshot', () => {
  test('Comprobar la UI del componente BusinessLine', () => {
    const businessLine = create(<BusinessLine />)
    expect(businessLine.toJSON()).toMatchSnapshot()
  })
})