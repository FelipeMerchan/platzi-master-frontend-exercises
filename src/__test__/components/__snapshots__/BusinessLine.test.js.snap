// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusinessLine Snapshot Comprobar la UI del componente BusinessLine 1`] = `
<article
  className="BusinessLine"
>
  <figure
    className="BusinessLine__imageContainer"
  >
    <img
      className="BusinessLine__image"
      width="200"
    />
  </figure>
  <div>
    <h3
      className="BusinessLine__name text-consultancy-color"
    />
    <p
      className="BusinessLine__description"
    >
      Lorem ipsum dolor sit amet, consectetuer aiscing elit.
    </p>
  </div>
</article>
`;
